File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Entities/Organization.cs
using Feedback.Analyzer.Domain.Common.Entities;

namespace Feedback.Analyzer.Domain.Entities;

/// <summary>
/// Represents an organization within the system.
/// </summary>
public class Organization : SoftDeletedEntity
{
    /// <summary>
    /// The unique identifier of the client associated with this organization.
    /// </summary>
    public Guid ClientId { get; set; }
    
    /// <summary>
    /// Gets or sets the  name of the Organization.
    /// </summary>
    public string Name { get; set; } = default!;
    
    /// <summary>
    ///  Gets or sets the  description of the Organization.
    /// </summary>
    public string Description { get; set; } = default!;

    /// <summary>
    /// Gets or sets a collection of Clients.
    /// </summary>
    public Client Client { get; set; } = default!;

    /// <summary>
    /// Gets or sets the collection of products.
    /// </summary>
    public IEnumerable<Product> Products { get; set; } = default!;
}

File: src/Insight.Box.FrontEnd/src/modules/organizations/models/Organization.ts
/*
 * Represents an organization
 */
export class Organization {
    /*
     * Organization id
     */
    public id: string;

    /*
     * Client id of the organization
     */
    public clientId: string;

    /*
     * Name of the organization
     */
    public name: string = '';

    /*
     * Description of the organization
     */
    public description: string = '';

    /*
     * Count of products in the organization
     */
    public productsCount: number;
}